pkg/controller/v1alpha2/applicationconfiguration/component.go

```diff
@@ -168,7 +173,7 @@ func (c *ComponentHandler) createControllerRevision(mt metav1.Object, obj runtim
 		Revision: nextRevision,
 		Data:     runtime.RawExtension{Object: curComp},
 	}
-	_, err := c.AppsClient.ControllerRevisions(mt.GetNamespace()).Create(context.Background(), &revision, metav1.CreateOptions{})
+	err := c.Client.Create(context.TODO(), &revision)
```

Comment:
what's the difference between TODO and Background?

Reply:
`Background` will never cancel, usually used in top-level, like main functions.
`Todo` used in an unclear context.